bepasty_cli/cli.py

```diff
@@ -50,20 +64,70 @@ def convert(self, value, param, ctx):
             self.fail('"%s" is not a valid lifetime.' % value, param, ctx)
 
 
+def setup_config_parser(override=None):
+    if not override:
+        override = dict()
+    config = ConfigParser()
+    config.add_section('bepasty-client-cli')
+    for k, v in CONFIG_DEFAULTS.items():
+        v = override.get(k, v)
+        config.set('bepasty-client-cli', k, str(v))
+    return config
+
+
+def write_configuration_file(config):
+    print('Writing configuration to "{}".'.format(CONFIG_FILENAME))
+    try:
+        os.makedirs(os.path.dirname(CONFIG_FILENAME), 750)
+    except OSError as exc:
+        if exc.errno != errno.EEXIST:
+            print(exc)
+            sys.exit(1)
+    with codecs.open(CONFIG_FILENAME, 'w', 'utf8') as fp:
+        config.write(fp)
```

Comment:
It is a common approach for the data a program produces (its purpose), but not for configuration files. You cannot tell vi, samba, or ncftp where to write their configuration to. Nowadays XDG handles this, so users have a chance to find files where they expect them.
A user who needs multiple "profiles" - very few do - can simply copy those after creating them and then supply them for usage with `-c`.